### ISSUE TITLE
Crop and/or downsample raw data before actual processing

### ISSUE THREAD

**Issue by crazyapril** [pytroll/satpy#1732]
## Feature Request

**Is your feature request related to a problem? Please describe.**
I'm using satpy to generate Himawari-8 true color images for a rather small region, but it seems to me that satpy would load and composite every bit of data users feed in, and that's very time & memory consuming. `Scene.crop` method can only be applied after true color data is fully loaded. Is there a way to crop bands before actual composite work?

**Describe the solution you'd like**
I think it would be good to add an option to `Scene.load` method that specifies area of interest. Satpy would crop raw data on the fly before actual processing. It may involves two steps: 1. filtering segment files (which currently can be controlled by user, but it would be nice for satpy to automatically select only needed files) 2. slice raw data to minimum required for a given area

**Comment by djhoese** [comment]
Could you please provide the code you are using that isn't working as you expect? What you are asking for, in principal, is how it is supposed to work. It may require a slight workaround for some special cases (like composites that need resampling to be generated), but overall we are using dask so none of the large data arrays are meant to be loaded until they are actually needed. There may be some readers where parsing information from the files may take longer and may require more memory than other formats, but there isn't much we can do about that. Which AHI format are you using (what reader)? There isn't much we can do about basic file parsing time. If you give us the files, we don't know we don't need it until we've opened it and parsed the metadata.

Before we used dask we used to have very basic functionality for filtering file loading based on a provided AreaDefinition, but we realized after switching to dask and xarray that this wasn't really needed. It also is very complicated to get right and complicates the readers in Satpy a lot.

So bottom line is, what code are you running and when does it use more memory or take longer than you expect? Is it at the end when you are actually computing things?
